docs/DeveloperGuide.md

```diff
@@ -1,22 +1,266 @@
+2.  Add a new constant with the message text. Example:
+```java
+NEW_MESSAGE("Your custom message text here");
+```
+3. Reference the new message in the `UI` class or any other relevant part of the application using `Messages.NEW_MESSAGE`
+
+
+
+### Parser
+
+The `Parser` class is a crucial component instantiated as soon as FindOurSEP is initialized. Its responsibility is to process the user’s input into commands and invoking the correct command object for the rest of the program.
+
+Some of its core features include:
+- Breaking down user input and extracting the relevant command and data for further processing.
+- Provide robust error handling when unknown command is received.
+- Validate data parsed in from external file (.CSV, .JSON, .TXT) sources. (Further details in the `Storage` class)
+
+Here is a class diagram highlighting the fundamental structure of the `Parser` class.
+
+![ParserClass](UML_Diagrams/ParserClass.drawio.svg)
+
+How `Parser` works:
+1. Whenever the user enters an input, the input will be directed to the `Parser` class's `parseInput()` method. 
+2. Within the method, the command will be extracted and the appropriate `XYZCommand` object (XYZCommand is a placeholder for various commands such as DeleteCommand, ListCommand, etc.) will be instantiated.
+3. Upon instantiation, the `XYZCommand` is prepared for execution. Each `XYZCommand` class, inheriting from the abstract `Command` class, has a `run()` method that executes its specific instructions.
+
+The sequence diagram below demonstrates the interactions within the `Parser` component when a user inputs the command: `add id/A1234567I gpa/5.0 p/{13,61,43}`.
+
+![ParserSequence](UML_Diagrams/ParserSequence.drawio.svg)
+
+The boolean return value of `parseInput()` indicates whether the user has chosen to continue or terminate the program. A `true` value keeps FindOurSEP running, while a `false` value ends the program.
+
+### Commands
+
+#### Add Command
 
-## Design & implementation
+#### Delete Command
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Delete Command removes an exisiting Student in the StudentList.
 
+![DeleteCommandSequence](./UML_Diagrams/deleteCommand.drawio.svg)
+
+#### Criteria Command
+
+Criteria Command sets a minimum GPA every student must acheieve before they can be allocated to a university.
+
+![CriteriaCommandSequence](./UML_Diagrams/CriteriaCommand.drawio.svg)
+
+#### Find Command
+
+#### Filter Command
+
+#### Stats Command
+
+#### ViewQuota Command
+
+
+#### List Command
+
+#### Stats Command
+
+The `StatCommand` class implements the `stats` command, which provides GPA-related statistics (average GPA or minimum GPA) for students associated with a specified university. The command syntax is `stats <stat_type> <UNI_INDEX>`, where `<stat_type>` can be `-avggpa` for average GPA or `-mingpa` for minimum GPA.
+
+![StatSequence](UML_Diagrams/StatSequence.drawio.svg)
```

Comment:
I think its better to standardize all the sequence diagrams for your commands. Either make the activation bar starts form the Command class directly or use dotted line first.